fastplotlib/layouts/_frame/_frame.py

```diff
@@ -0,0 +1,184 @@
+        return self._output
+
+    def render(self):
+        """render call implemented in subclass"""
+        raise NotImplemented
+
+    def _autoscale_init(self, maintain_aspect: bool):
+        """autoscale function that is called only during show()"""
+        if hasattr(self, "_subplots"):
+            for subplot in self:
+                if maintain_aspect is None:
+                    _maintain_aspect = subplot.camera.maintain_aspect
+                else:
+                    _maintain_aspect = maintain_aspect
+                subplot.auto_scale(maintain_aspect=_maintain_aspect, zoom=0.95)
+        else:
+            if maintain_aspect is None:
+                maintain_aspect = self.camera.maintain_aspect
+            self.auto_scale(maintain_aspect=maintain_aspect, zoom=0.95)
+
+    def start_render(self):
+        """start render cycle"""
+        self.canvas.request_draw(self.render)
+        self.canvas.set_logical_size(*self._starting_size)
+
+    def show(
+            self,
+            autoscale: bool = True,
+            maintain_aspect: bool = None,
+            toolbar: bool = True,
+            sidecar: bool = False,
+            sidecar_kwargs: dict = None,
+            add_widgets: list = None,
+    ):
+        """
+        Begins the rendering event loop and shows the plot in the desired output context (jupyter, qt or glfw).
+
+        Parameters
+        ----------
+        autoscale: bool, default ``True``
+            autoscale the Scene
+
+        maintain_aspect: bool, default ``True``
+            maintain aspect ratio
+
+        toolbar: bool, default ``True``
+            show toolbar
+
+        sidecar: bool, default ``True``
+            display plot in a ``jupyterlab-sidecar``, only for jupyter output context
+
+        sidecar_kwargs: dict, default ``None``
+            kwargs for sidecar instance to display plot
+            i.e. title, layout
+
+        add_widgets: list of widgets
+            a list of ipywidgets or QWidget that are vertically stacked below the plot
+
+        Returns
+        -------
+        OutputContext
+            In jupyter, it will display the plot in the output cell or sidecar
+
+            In Qt, it will display the Plot, toolbar, etc. as stacked widget, use `Plot.widget` to access it.
+        """
+
+        # show was already called, return existing output context
+        if self._output is not None:
+            return self._output
+
+        self.start_render()
+
+        if sidecar_kwargs is None:
+            sidecar_kwargs = dict()
+
+        if add_widgets is None:
+            add_widgets = list()
+
+        # flip y axis if ImageGraphics are present
+        if hasattr(self, "_subplots"):
+            for subplot in self:
+                for g in subplot.graphics:
+                    if isinstance(g, ImageGraphic):
+                        subplot.camera.local.scale_y = -1
+                        break
```

Comment:
closes #268

Reply:
One thing I notice is that this logic is not implemented for imagewidget

Reply:
ImageWidget uses a Gridplot so it's not necessary